libs/build_esmf.sh

```diff
@@ -113,7 +113,8 @@ case $MPI in
     export ESMF_COMM=${STACK_esmf_comm:-"mpich3"}
     ;;
   cray-mpich )
-    export ESMF_COMM=${STACK_esmf_comm:-"mpi"}
+    export ESMF_OS="Linux"
+    export ESMF_COMM=${STACK_esmf_comm:-"mpich3"}
```

Comment:
I think this is going to break Gaea.
@climbfuji Can you please verity?

Reply:
Yes, this will not work.

Reply:
What part will break it? `ESMF_COMM` or `ESMF_OS` (or both)?

Reply:
Keep the "export ESMF_COMM=${STACK_esmf_comm:-"mpi"}
Will break the wcoss2. (export in the conf does not work)
We have to finger out a way.

Reply:
Two system settings of cray-mpich are different.

Reply:
Setting the environment variable should work. 

Let's figure out why setting `STACK_esmf_comm` doesn't work because it should.

Reply:
@Hang-Lei-NOAA 
Please add
`export STACK_esmf_comm="mpich3"` in `config_wcoss2.sh` 
This **should** work.

Reply:
I tryed that is not working. so I changed it.

Reply:
Kyle, could you please try it again?

Reply:
The ESMF_OS line also needs to be removed, and possibly an option to overwrite the default setting "Unicos" needs to be added.

Reply:
I would like to, but I'm having a problem with my WCOSS account and I can't login. I'm waiting for a response from the help desk.

Reply:
Try printing out the contents of the variable in the build script and see if it's set.

Reply:
After trying different combinations.
The issue is on ESMF_OS. It has to be there.
Taking Rahul's suggestion by using" export ESMF_OS=${STACK_esmf_os:-$host}" works

Reply:
Change default ESMF_COMM back to mpi for gaea, and export it to mpich3 in config file for wcoss2.

Reply:
@climbfuji Could you please try if adding "export ESMF_OS=${STACK_esmf_os:-$host}" will affact the gaea? Thanks

Reply:
I am trying this now.